server/handler/details.go

```diff
@@ -109,43 +108,46 @@ func (h *Details) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
 	}
 
 	var data struct {
-		Error       error
-		Result      *common.Result
-		PullRequest *github.PullRequest
-		User        string
-		PolicyURL   string
+		Error            error
+		IsTemporaryError bool
+		Result           *common.Result
+		PullRequest      *github.PullRequest
+		User             string
+		PolicyURL        string
 	}
 
 	data.PullRequest = pr
 	data.User = user
 
 	config, err := h.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	data.PolicyURL = getPolicyURL(pr, config)
-
 	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration at ref=%s", config.Ref))
+		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration: %s", config))
 		return h.render(w, data)
 	}
 
-	if config.Missing() {
-		data.Error = errors.New(config.Description())
-		return h.render(w, data)
-	}
+	evaluator, err := h.Base.ValidateFetchedConfig(ctx, prctx, client, config, common.TriggerAll)
 
-	if config.Invalid() {
-		data.Error = errors.WithMessage(config.Error, config.Description())
+	if err != nil {
+		data.Error = err
 		return h.render(w, data)
 	}
 
-	evaluator, err := policy.ParsePolicy(config.Config)
-	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("invalid policy at ref \"%s\"", config.Ref))
+	if evaluator == nil {
+		data.Error = errors.Errorf("Unable to evaluate nil evaluator: %s", config.Description())
```

Comment:
```suggestion
		// config.Description() includes a good message for missing policies
		data.Error = errors.New(config.Description())
```

`nil evaluator` is a pretty low-level implementation detail for a UI message and it sounds like the existing description is good enough to tell people what the problem is. If you do want a prefix, I think the one you had before was better ("Unable to evaluate").